### DISCUSSION TITLE
What should be the git repo setup for the pocketbase project?

### DISCUSSION THREAD

**Discussion by michaltaberski** [pocketbase/pocketbase#1158] [Q&A] [answered]
The question might sound silly at first, but I was looking for an answer without success yet. 

I would like to setup github repo FE + BE using pocketbase for the BE for the first time. I wonder if the `pocketbase` binary belongs to the repo or it should be in `.gitignore` and I should add a script downloading it each time 🤔 .

Should I think about `pocketbase` more like about `node_modules` in FE app? Which generated files should be committed and what should be ignored?

**Comment by clemorej** [comment]
You should exclude the pocketbase because you don't really track the changes within the binary. Just add a note somewhere in your app version of  pocketbase you're using.

**Reply by nicosql** [reply]
that isn't a great suggestion for a project that hasn't 1.0'd yet.  I wouldn't risk it.

**Comment by ganigeorgiev** [comment, marked as answer]
@michaltaberski If you use the prebuilt executable and want your git repo to be "ready out of the box" then you can commit the binary in your repo. The binary is not that big (compressed is ~12mb) so it shouldn't matter too much. This also has another benefit and ensures that when the repo is cloned, the user will use the PocketBase version you've already tested and know that it works.

_In the next release, the prebuilt executable will also provide seamless automigrations that will be generated in a `pb_migrations` directory and you'll be able to safely commit it to your repo and share/store your collections configuration (including default settings, custom data imports, etc.)._

**Reply by benallfree** [reply]
I had a thought that maybe a pocketbase-cli npm package would be useful, and all it would do is install the correct binary. 

esbuild is a good example of this.

**Reply by ganigeorgiev** [reply]
@benallfree I'm not sure if there will be any benefit of doing that, especially for SPA or prerendered client applications, since you will not need the node dependency in those cases when deploying the application. For esbuild, sass, and other cli tools it make sense to be distributed this way because they are part of the build chain.

**Reply by benallfree** [reply]
I consider devDependencies to be for anything used during dev. prettier, chokidar, nodemon, tsx, concurrently are examples I use frequently that aren’t part of the build chain. 

Publishing a bin package would allow devs to lock the PocketBase bin version and put it inside an npm:run script command. 

A similar case could be made for prod. I might add pocketbase-cli as a prod dependency depending on how I was planning to deploy the backend.

**Reply by ganigeorgiev** [reply]
@benallfree I'm not sure that I understand the use case here. All listed examples are part of the build/dev process, even if they aren't explicitly executed. And in all cases you need them mainly during local development, which is not the case with PocketBase, since you have to deploy it and run it also on production (things could get even more complicated if the PocketBase executable must be hosted on a different server in case vercel, cloudflare workers, etc. is used).

> Publishing a bin package would allow devs to lock the PocketBase bin version and put it inside an npm:run script command.

But that's the thing. It will require to have node to execute it, which is fine during local development, but on production you may not need it. And now instead of the traditional command line execution, users will have to go through additional steps to just execute the binary. It'll also require additional install/run documentation, which could be confusing to some users.

**Reply by benallfree** [reply]
You’re right it wouldn’t fit every use case and deployment scenario. 

The case for dev: It would make node development more convenient and shareable since the bin could be treated as a node dependency and locked in the repo. It would always install the correct architecture with no ambiguity. It would allow something like `npx pocketbase` for quick stand-ups and experiments always using the latest version. For me as a full time node dev, all of these points are compelling reasons to have a bin or cli package. 

The case for prod: It *might*, in some deployment scenarios, be nice for node developers to treat PocketBase as a prod dependency. If they are making a node-based backend that runs along side and depends upon PocketBase, then PocketBase may or may not even be exposed to the end user. And if it is, it might be proxied. through node. I have this scenario with PocketHost, except my case is more complicated because I need to run all the versions of PocketBase not just one specific version. But the point is, it does feel like PocketBase is a prod dependency more than a stand-alone backend. 

Anyway I guess someone who wants to maintain it can always create and maintain a cli bin package. Those are just my personal reasons for thinking it’s a good idea :)

**Reply by nicosql** [reply]
yea node isn't the de facto use case anymore, thank god
